OTEP.md

```diff
@@ -32,7 +32,473 @@ Using a configuration model or configuration file, users could configure all opt
 * Versioning: needs to be able to version stability while evolving
 * (stretch) The file format can be validated client side.
 
-## Internal details (TBD)
+## Internal details
+
+In order to provide a minimal API surface area, implementations *MUST* support the following methods.
+
+### LoadAndValidateConfigurationFromFile(filename, format)
+
+An API called `LoadAndValidateConfigurationFromFile` receives a string parameter indicating the file containing the configuration to be loaded. An optional format parameter may be provided to indicate the format that this configuration uses. The default value for this parameter is `yaml`. The method returns a `Configuration` model that has been validated. This API *MAY* return an error or raise an exception, whichever is idiomatic to the implementation for the following reasons:
+
+* file doesn't exist or is invalid
+* configuration loaded is invalid
+
+#### Python LoadAndValidateConfigurationFromFile example
+
+```python
+
+filename = "./config.yaml"
+
+try:
+  cfg = opentelemetry.LoadAndValidateConfigurationFromFile(filename)
+except Exception as e:
+  print(e)
+
+filename = "./config.ini"
+
+try:
+  cfg = opentelemetry.LoadAndValidateConfigurationFromFile(filename, format="ini")
+except Exception as e:
+  raise e
+
+```
+
+#### Go LoadAndValidateConfigurationFromFile example
+
+```go
+
+filename := "./config.yaml"
+cfg, err := otel.LoadAndValidateConfigurationFromFile(filename)
+if err != nil {
+  return err
+}
+
+filename := "./config.json"
+cfg, err := otel.LoadAndValidateConfigurationFromFile(filename, otelconfig.WithFormat("json"))
+if err != nil {
+  return err
+}
+
+```
+
+### Configuration model
+
+To allow SDKs and instrumentation libraries to load configuration without having to implement the parsing logic, a `Configuration` model *MUST* be provided by implementations. This object:
+
+* has already been parsed from a file or data structure
+* has been validated
+
+(TBD what methods should this configuration model make available for SDKs/instrumentations?)
+
+### Additional interface: LoadAndValidateConfiguration
+
+Each language implementation supporting OpenTelemetry *MAY* support parsing a data structure instead of a file to produce a model. This allows implementations to provide a configuration interface without the expectation on users to load a configuration file. The following demonstrates how Python and Go may provide a configuration interface to accomplish this:
+
+#### Python LoadAndValidateConfiguration example
+
+```python
+opentelemetry.LoadAndValidateConfiguration(
```

Comment:
nit: Once you have the data structure, does it still make sense to prefix `LoadAnd`? Could this be `ValidateConfiguration`? The `Load` part is not clear to me. Does it exist to indicate it returns a Config object? If so would `ValidateAndRetrieveConfiguration` be more clear?

Reply:
Maybe something along the lines of ParseAndValidate?

Reply:
ParseAndValidate sounds much better, but I don't think this is really a concern.